compiler/Symbolizer.cpp

```diff
@@ -619,11 +636,20 @@ void Symbolizer::visitBitCastInst(BitCastInst &I) {
 
 void Symbolizer::visitTruncInst(TruncInst &I) {
   IRBuilder<> IRB(&I);
-  auto trunc = buildRuntimeCall(
-      IRB, runtime.buildTrunc,
+
+  SymbolicComputation symbolicComputation;
```

Comment:
Could you add a line that returns early if `getSymbolicExpression(I.getOperand(0)) == nullptr`? We don't need to emit code if the argument is known to be concrete at compile time. `buildRuntimeCall` checks implicitly, but now that we use `forceBuildRuntimeCall` we have to perform the optimization manually :frowning_face: